.github/workflows/ui.yaml

```
@@ -0,0 +1,147 @@
+name: "UI Build"
+on:
+  push:
+    tags:
+      - '[0-9]+.[0-9]+.[0-9]+'
+      - '[0-9]+.[0-9]+.[0-9]+-rc[0-9]+'
+    branches: [ $default-branch ]
+    # Path filters aren't evaluated for tags - https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore
+    paths:
+      - '.nvmrc'
+      - 'services/ui/**'
+  pull_request:
+    paths:
+      - '.nvmrc'
+      - 'services/ui/**'
+    branches: [ $default-branch ]
+env:
+  REGISTRY: ghcr.io/noaa-gsl/unified-graphics/ui
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version-file: '.nvmrc'
+          cache: 'npm'
+          cache-dependency-path: 'services/ui/package-lock.json'
+      - name: Install dependencies
+        working-directory: services/ui
+        run: npm ci
+      - name: Lint
+        working-directory: services/ui
+        run: npm run lint
+  check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version-file: '.nvmrc'
+          cache: 'npm'
+          cache-dependency-path: 'services/ui/package-lock.json'
+      - name: Install dependencies
+        working-directory: services/ui
+        run: npm ci
+      - name: Check
+        working-directory: services/ui
+        run: npm run check
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version-file: '.nvmrc'
+          cache: 'npm'
+          cache-dependency-path: 'services/ui/package-lock.json'
+      - name: Install dependencies
+        working-directory: services/ui
+        run: npm ci
+      - name: Test
+        working-directory: services/ui
+        run: |
+          npx playwright install-deps
+          npm run test
+  build:
+    runs-on: ubuntu-latest
+    needs: [lint, check, test]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Extract branch/tag name
+        shell: bash
+        run: |
+          if [[ "${GITHUB_EVENT_NAME}" == "pull_request" ]]; then
+            # PR build
+            echo "BRANCH=${GITHUB_HEAD_REF}" >> $GITHUB_ENV
+          elif [[ "${GITHUB_EVENT_NAME}" == "push" ]]; then
+            # Handle differences between branches/tags
+            if [[ "${GITHUB_REF}" == *"heads"* ]]; then
+              # Branch build
+              echo "BRANCH=${GITHUB_REF#refs/heads/}" >> $GITHUB_ENV
+            elif [[ "${GITHUB_REF}" == *"tags"* ]]; then
+              # Tag build
+              echo "BRANCH=${GITHUB_REF#refs/tags/}" >> $GITHUB_ENV
```

This is not a big deal, but if you’re following the [Community Bash Styleguide](https://github.com/esheehan-gsl/community_bash_style_guide) or the [Google Shell Styleguide](https://google.github.io/styleguide/shellguide.html#s5.6-variable-expansion) I think these should all be redirecting to `"${GITHUB_ENV}"` like this

```suggestion
          if [[ "${GITHUB_EVENT_NAME}" == "pull_request" ]]; then
            # PR build
            echo "BRANCH=${GITHUB_HEAD_REF}" >> "${GITHUB_ENV}"
          elif [[ "${GITHUB_EVENT_NAME}" == "push" ]]; then
            # Handle differences between branches/tags
            if [[ "${GITHUB_REF}" == *"heads"* ]]; then
              # Branch build
              echo "BRANCH=${GITHUB_REF#refs/heads/}" >> "${GITHUB_ENV}"
            elif [[ "${GITHUB_REF}" == *"tags"* ]]; then
              # Tag build
              echo "BRANCH=${GITHUB_REF#refs/tags/}" >> "${GITHUB_ENV}"
```

We haven’t committed to a Bash styleguide (no linting or anything), so I don’t see a reason to insist on this, but this is the convention I followed for the Git hooks, so I thought I’d just mention it.

---

Great point. Thanks for catching that.